hkmc2/shared/src/main/scala/hkmc2/syntax/Tree.scala

```diff
@@ -192,20 +192,24 @@ enum Tree extends AutoLocated:
       LetLike(letLike, id, S(App(Ident(nme.init), Tup(id :: r :: Nil))), bodo).withLocOf(this).desugared
     case _ => this
   
-  /** S(true) means eager spread, S(false) means lazy spread, N means no spread. */
-  def asParam: Opt[(Opt[Bool], Ident, Opt[Tree])] = this match
+  /** 
+   * S(true) means eager spread, S(false) means lazy spread, N means no spread.
+   * ctx means the the param list is modified by `using`.
+   */
+  def asParam(inUsing: Bool): Opt[(Opt[Bool], Ident, Opt[Tree])] = this match
     case und: Under => S(N, new Ident("_").withLocOf(und), N)
+    case id: Ident if inUsing => S(N, Ident(""), S(id))
```

Comment:
Would be good to document this.
```suggestion
    // * In `using` clauses, identifiers are understood as type names for unnamed contextual parameters:
    case id: Ident if inUsing => S(N, Ident(""), S(id))
```